[TESTING]: Admin logout token revocation tests failing with 404 errors since PR 4371

**Title:** [TESTING]: Admin logout token revocation tests failing with 404 errors since [PR 4371](https://github.com/IBM/mcp-context-forge/pull/4371)

**Labels:** testing, bug, triage

**Body content:**

### ✅ Test Summary
All 7 tests in `tests/unit/mcpgateway/test_admin_logout_token_revocation.py` are failing with 404 errors when attempting to access the `/admin/logout` endpoint.

---

### 🧪 Test Type
Choose one or more applicable categories:

- [x] Unit tests (Python)
- [ ] Integration / end-to-end tests
- [ ] Manual UI testing (admin panel)
- [ ] Transport coverage (HTTP, WebSocket, SSE, stdio)
- [ ] Federation / multi-gateway testing
- [ ] Configuration / environment-specific behavior
- [ ] Other (explain below)

---

### 🧬 Scope & Affected Components
Select what this test covers or validates:

- [x] `mcpgateway` core (API logic, handlers)
- [ ] Admin UI (HTMX / Alpine / Tailwind)
- [ ] Tool/Resource/Prompt logic
- [ ] Federation sync/discovery
- [ ] Server interactions or SSE
- [x] Auth / JWT / security flows
- [ ] Observability (logging, metrics)
- [ ] Makefile, shell scripts or CLI
- [ ] Containerized setup (Docker/Podman)
- [ ] Other (explain below)

---

### 📋 Acceptance Criteria
What should pass or be verified?

- [x] All relevant assertions are covered
- [x] No side-effects or regressions observed
- [ ] Confirmed in multiple environments (if needed)
- [x] Edge cases and error handling tested
- [x] Logs and output are clean and expected

---

### 📓 Additional Context & Steps

**Root Cause:**
The test file creates `TestClient(app)` directly without using the required `main_app_with_admin_api` fixture. Since admin routes are feature-flagged behind `MCPGATEWAY_ADMIN_API_ENABLED` (which defaults to `false` in tests per `tests/conftest.py:89`), the `/admin/logout` endpoint is not mounted, resulting in 404 responses.

**Failing Tests:**
1. `test_post_revokes_token_in_blocklist`
2. `test_post_without_jwt_cookie_does_not_call_blocklist`
3. `test_post_with_invalid_jwt_cookie_does_not_block_logout`
4. `test_post_payload_without_jti_does_not_revoke`
5. `test_post_blocklist_failure_does_not_block_logout`
6. `test_post_with_jwt_cookie_but_no_csrf_token_is_rejected`
7. `test_get_logout_is_exempt_from_csrf`

**Expected Behavior:**
Tests should successfully access the `/admin/logout` endpoint and verify token revocation behavior.

**Actual Behavior:**
All requests to `/admin/logout` return 404 Not Found.

**Solution:**
Update the test file to use the `main_app_with_admin_api` fixture (see examples in `tests/unit/mcpgateway/test_well_known.py:26-28` and `tests/unit/mcpgateway/test_admin_catalog_htmx.py:23-25`).

**Affected Files:**
- `tests/unit/mcpgateway/test_admin_logout_token_revocation.py` (lines 90, 107, 123, 140, 158, 182, 205)
- `mcpgateway/main.py:11723-11731` (admin route mounting logic)
- `tests/conftest.py:89` (default test config)
- `tests/conftest.py:350-420` (required fixture)

**Reproduction:**
```bash
# Run the failing test suite
pytest tests/unit/mcpgateway/test_admin_logout_token_revocation.py

# Get failure logs:
FFFFFFF                                                                                                                                                                                                                                [100%]
================================================================================================================== FAILURES ==================================================================================================================
____________________________________________________________________________________ TestAdminLogoutTokenRevocation.test_post_revokes_token_in_blocklist _____________________________________________________________________________________

self = <tests.unit.mcpgateway.test_admin_logout_token_revocation.TestAdminLogoutTokenRevocation object at 0x11108f6b0>, disable_admin_csrf = None

    def test_post_revokes_token_in_blocklist(self, disable_admin_csrf):
        token, payload = _build_admin_jwt()

        with (
            patch("mcpgateway.admin.verify_jwt_token_cached", new_callable=AsyncMock) as mock_verify,
            patch("mcpgateway.services.token_blocklist_service.get_token_blocklist_service") as mock_get_service,
        ):
            mock_verify.return_value = payload
            mock_blocklist = MagicMock()
            mock_blocklist.revoke_token.return_value = True
            mock_get_service.return_value = mock_blocklist

            client = TestClient(app, follow_redirects=False)
            response = client.post("/admin/logout", cookies={"jwt_token": token})

>           assert response.status_code in (302, 303, 307, 200)
E           assert 404 in (302, 303, 307, 200)
E            +  where 404 = <Response [404 Not Found]>.status_code

tests/unit/mcpgateway/test_admin_logout_token_revocation.py:93: AssertionError
------------------------------------------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------------------------------------------
2026-04-27T15:23:22 - httpx - INFO - HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------
INFO     httpx:_client.py:1025 HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
____________________________________________________________________________ TestAdminLogoutTokenRevocation.test_post_without_jwt_cookie_does_not_call_blocklist _____________________________________________________________________________

self = <tests.unit.mcpgateway.test_admin_logout_token_revocation.TestAdminLogoutTokenRevocation object at 0x1128a13a0>, disable_admin_csrf = None

    def test_post_without_jwt_cookie_does_not_call_blocklist(self, disable_admin_csrf):
        with patch("mcpgateway.services.token_blocklist_service.get_token_blocklist_service") as mock_get_service:
            mock_blocklist = MagicMock()
            mock_get_service.return_value = mock_blocklist

            client = TestClient(app, follow_redirects=False)
            response = client.post("/admin/logout")

>           assert response.status_code in (302, 303, 307, 200)
E           assert 404 in (302, 303, 307, 200)
E            +  where 404 = <Response [404 Not Found]>.status_code

tests/unit/mcpgateway/test_admin_logout_token_revocation.py:110: AssertionError
------------------------------------------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------------------------------------------
2026-04-27T15:23:22 - httpx - INFO - HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------
INFO     httpx:_client.py:1025 HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
___________________________________________________________________________ TestAdminLogoutTokenRevocation.test_post_with_invalid_jwt_cookie_does_not_block_logout ___________________________________________________________________________

self = <tests.unit.mcpgateway.test_admin_logout_token_revocation.TestAdminLogoutTokenRevocation object at 0x1128a2f90>, disable_admin_csrf = None

    def test_post_with_invalid_jwt_cookie_does_not_block_logout(self, disable_admin_csrf):
        with (
            patch("mcpgateway.admin.verify_jwt_token_cached", new_callable=AsyncMock) as mock_verify,
            patch("mcpgateway.services.token_blocklist_service.get_token_blocklist_service") as mock_get_service,
        ):
            mock_verify.side_effect = Exception("Invalid token")
            mock_blocklist = MagicMock()
            mock_get_service.return_value = mock_blocklist

            client = TestClient(app, follow_redirects=False)
            response = client.post("/admin/logout", cookies={"jwt_token": "not.a.valid.jwt"})

>           assert response.status_code in (302, 303, 307, 200)
E           assert 404 in (302, 303, 307, 200)
E            +  where 404 = <Response [404 Not Found]>.status_code

tests/unit/mcpgateway/test_admin_logout_token_revocation.py:125: AssertionError
------------------------------------------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------------------------------------------
2026-04-27T15:23:22 - httpx - INFO - HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------
INFO     httpx:_client.py:1025 HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
________________________________________________________________________________ TestAdminLogoutTokenRevocation.test_post_payload_without_jti_does_not_revoke ________________________________________________________________________________

self = <tests.unit.mcpgateway.test_admin_logout_token_revocation.TestAdminLogoutTokenRevocation object at 0x1128a3410>, disable_admin_csrf = None

    def test_post_payload_without_jti_does_not_revoke(self, disable_admin_csrf):
        token, payload = _build_admin_jwt(include_jti=False)
        assert "jti" not in payload

        with (
            patch("mcpgateway.admin.verify_jwt_token_cached", new_callable=AsyncMock) as mock_verify,
            patch("mcpgateway.services.token_blocklist_service.get_token_blocklist_service") as mock_get_service,
        ):
            mock_verify.return_value = payload
            mock_blocklist = MagicMock()
            mock_get_service.return_value = mock_blocklist

            client = TestClient(app, follow_redirects=False)
            response = client.post("/admin/logout", cookies={"jwt_token": token})

>           assert response.status_code in (302, 303, 307, 200)
E           assert 404 in (302, 303, 307, 200)
E            +  where 404 = <Response [404 Not Found]>.status_code

tests/unit/mcpgateway/test_admin_logout_token_revocation.py:143: AssertionError
------------------------------------------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------------------------------------------
2026-04-27T15:23:22 - httpx - INFO - HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------
INFO     httpx:_client.py:1025 HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
______________________________________________________________________________ TestAdminLogoutTokenRevocation.test_post_blocklist_failure_does_not_block_logout ______________________________________________________________________________

self = <tests.unit.mcpgateway.test_admin_logout_token_revocation.TestAdminLogoutTokenRevocation object at 0x1128a31a0>, disable_admin_csrf = None

    def test_post_blocklist_failure_does_not_block_logout(self, disable_admin_csrf):
        token, payload = _build_admin_jwt()

        with (
            patch("mcpgateway.admin.verify_jwt_token_cached", new_callable=AsyncMock) as mock_verify,
            patch("mcpgateway.services.token_blocklist_service.get_token_blocklist_service") as mock_get_service,
        ):
            mock_verify.return_value = payload
            mock_blocklist = MagicMock()
            mock_blocklist.revoke_token.side_effect = Exception("Database unavailable")
            mock_get_service.return_value = mock_blocklist

            client = TestClient(app, follow_redirects=False)
            response = client.post("/admin/logout", cookies={"jwt_token": token})

>           assert response.status_code in (302, 303, 307, 200)
E           assert 404 in (302, 303, 307, 200)
E            +  where 404 = <Response [404 Not Found]>.status_code

tests/unit/mcpgateway/test_admin_logout_token_revocation.py:161: AssertionError
------------------------------------------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------------------------------------------
2026-04-27T15:23:22 - httpx - INFO - HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------
INFO     httpx:_client.py:1025 HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
______________________________________________________________________________ TestAdminLogoutDenyPaths.test_post_with_jwt_cookie_but_no_csrf_token_is_rejected ______________________________________________________________________________

self = <tests.unit.mcpgateway.test_admin_logout_token_revocation.TestAdminLogoutDenyPaths object at 0x1128a3680>

    def test_post_with_jwt_cookie_but_no_csrf_token_is_rejected(self):
        """Cookie auth + state-changing POST without a CSRF token must return 403.

        This protects against cross-site-forced-logout attacks. No
        ``disable_admin_csrf`` fixture here — the real ``enforce_admin_csrf``
        dependency runs and must reject the request.
        """
        token, _ = _build_admin_jwt()

        with patch("mcpgateway.services.token_blocklist_service.get_token_blocklist_service") as mock_get_service:
            mock_blocklist = MagicMock()
            mock_get_service.return_value = mock_blocklist

            client = TestClient(app, follow_redirects=False)
            response = client.post("/admin/logout", cookies={"jwt_token": token})

>           assert response.status_code == 403
E           assert 404 == 403
E            +  where 404 = <Response [404 Not Found]>.status_code

tests/unit/mcpgateway/test_admin_logout_token_revocation.py:184: AssertionError
------------------------------------------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------------------------------------------
2026-04-27T15:23:22 - httpx - INFO - HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------
INFO     httpx:_client.py:1025 HTTP Request: POST http://testserver/admin/logout "HTTP/1.1 404 Not Found"
________________________________________________________________________________________ TestAdminLogoutDenyPaths.test_get_logout_is_exempt_from_csrf ________________________________________________________________________________________

self = <tests.unit.mcpgateway.test_admin_logout_token_revocation.TestAdminLogoutDenyPaths object at 0x1128a3b00>

    def test_get_logout_is_exempt_from_csrf(self):
        """GET /admin/logout supports OIDC front-channel logout and is exempt from CSRF.

        Per the OIDC Front-Channel Logout 1.0 spec the IdP issues a GET; we
        must accept it without a CSRF token.
        """
        token, payload = _build_admin_jwt()

        with (
            patch("mcpgateway.admin.verify_jwt_token_cached", new_callable=AsyncMock) as mock_verify,
            patch("mcpgateway.services.token_blocklist_service.get_token_blocklist_service") as mock_get_service,
        ):
            mock_verify.return_value = payload
            mock_blocklist = MagicMock()
            mock_blocklist.revoke_token.return_value = True
            mock_get_service.return_value = mock_blocklist

            client = TestClient(app, follow_redirects=False)
            response = client.get(
                "/admin/logout",
                cookies={"jwt_token": token},
                headers={"accept": "application/json"},
            )

>           assert response.status_code in (200, 302, 303, 307)
E           assert 404 in (200, 302, 303, 307)
E            +  where 404 = <Response [404 Not Found]>.status_code

tests/unit/mcpgateway/test_admin_logout_token_revocation.py:212: AssertionError
------------------------------------------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------------------------------------------
2026-04-27T15:23:22 - httpx - INFO - HTTP Request: GET http://testserver/admin/logout "HTTP/1.1 404 Not Found"
------------------------------------------------------------------------------------------------------------- Captured log call --------------------------------------------------------------------------------------------------------------
INFO     httpx:_client.py:1025 HTTP Request: GET http://testserver/admin/logout "HTTP/1.1 404 Not Found"
============================================================================================================== warnings summary ==============================================================================================================
tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutTokenRevocation::test_post_revokes_token_in_blocklist
tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutTokenRevocation::test_post_with_invalid_jwt_cookie_does_not_block_logout
tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutTokenRevocation::test_post_payload_without_jti_does_not_revoke
tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutTokenRevocation::test_post_blocklist_failure_does_not_block_logout
tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutDenyPaths::test_post_with_jwt_cookie_but_no_csrf_token_is_rejected
tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutDenyPaths::test_get_logout_is_exempt_from_csrf
  /Users/brian/dev/github.com/IBM/proj_mcp-context-forge/mcp-context-forge_main/.venv/lib/python3.12/site-packages/starlette/testclient.py:445: DeprecationWarning: Setting per-request cookies=<...> is being deprecated, because the expected behaviour on cookie persistence is ambiguous. Set cookies directly on the client instance instead.
    return super().request(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================================== short test summary info ===========================================================================================================
FAILED tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutTokenRevocation::test_post_revokes_token_in_blocklist - assert 404 in (302, 303, 307, 200)
FAILED tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutTokenRevocation::test_post_without_jwt_cookie_does_not_call_blocklist - assert 404 in (302, 303, 307, 200)
FAILED tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutTokenRevocation::test_post_with_invalid_jwt_cookie_does_not_block_logout - assert 404 in (302, 303, 307, 200)
FAILED tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutTokenRevocation::test_post_payload_without_jti_does_not_revoke - assert 404 in (302, 303, 307, 200)
FAILED tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutTokenRevocation::test_post_blocklist_failure_does_not_block_logout - assert 404 in (302, 303, 307, 200)
FAILED tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutDenyPaths::test_post_with_jwt_cookie_but_no_csrf_token_is_rejected - assert 404 == 403
FAILED tests/unit/mcpgateway/test_admin_logout_token_revocation.py::TestAdminLogoutDenyPaths::test_get_logout_is_exempt_from_csrf - assert 404 in (200, 302, 303, 307)
7 failed, 6 warnings in 1.09s


# Expected output: 7 failures with 404 errors
# After fix: 7 passing tests
```

---

### 🧠 Environment Info (if manual testing)

| Key              | Value                        |
| ---------------- | ---------------------------- |
| Gateway version  | `main` (post PR 4371)        |
| Python version   | `3.11+`                      |
| Transport tested | `http`                       |
| OS / Platform    | `All`                        |
| Container        | `N/A`                        |

---

### 📎 Related PRs / Issues (optional)

- PR #4371 (introduced the regression)
- Related to admin API feature flag implementation

--- a/.secrets.baseline
+++ b/.secrets.baseline
@@ -3,7 +3,7 @@
     "files": "(?x)( package-lock\\.json$ |Cargo\\.lock$ |uv\\.lock$ |go\\.sum$ |mcpgateway/sri_hashes\\.json$ )|^.secrets.baseline$",
     "lines": null
   },
-  "generated_at": "2026-04-27T14:14:15Z",
+  "generated_at": "2026-04-27T15:49:27Z",
   "plugins_used": [
     {
       "name": "AWSKeyDetector"

--- a/.secrets.baseline
+++ b/.secrets.baseline
@@ -3,7 +3,7 @@
     "files": "(?x)( package-lock\\.json$ |Cargo\\.lock$ |uv\\.lock$ |go\\.sum$ |mcpgateway/sri_hashes\\.json$ )|^.secrets.baseline$",
     "lines": null
   },
-  "generated_at": "2026-04-27T15:49:27Z",
+  "generated_at": "2026-04-27T15:55:30Z",
   "plugins_used": [
     {
       "name": "AWSKeyDetector"

--- a/tests/unit/mcpgateway/test_admin_logout_token_revocation.py
+++ b/tests/unit/mcpgateway/test_admin_logout_token_revocation.py
@@ -15,10 +15,12 @@ regressions are caught.
 """
 
 # Standard
+from collections.abc import Generator
 from datetime import datetime, timedelta, timezone
+from typing import Any
+
+from fastapi import FastAPI
 from unittest.mock import AsyncMock, MagicMock, patch
-import importlib
-import os
 import uuid
 
 # Third-Party
@@ -29,50 +31,11 @@ from fastapi.testclient import TestClient
 # First-Party
 from mcpgateway.admin import enforce_admin_csrf
 from mcpgateway.config import settings
-from mcpgateway.main import app
 
 
-@pytest.fixture(scope="module", autouse=True)
-def enable_admin_api():
-    """Enable admin API for logout tests.
-
-    The admin router is only registered when MCPGATEWAY_ADMIN_API_ENABLED=True.
-    This fixture sets the environment variable before the app is initialized,
-    clears the settings cache, and forces a reload of the main module to pick
-    up the setting.
-    """
-    # Store original value
-    original_value = os.environ.get("MCPGATEWAY_ADMIN_API_ENABLED")
-
-    # Set environment variable
-    os.environ["MCPGATEWAY_ADMIN_API_ENABLED"] = "true"
-
-    # Clear the settings cache so get_settings() will create a new instance
-    from mcpgateway.config import get_settings
-
-    get_settings.cache_clear()
-
-    # Force reload of config and main modules to pick up the setting
-    import mcpgateway.config
-    import mcpgateway.main
-
-    importlib.reload(mcpgateway.config)
-    importlib.reload(mcpgateway.main)
-
-    # Update the app reference in this module to use the reloaded app
-    global app
-    app = mcpgateway.main.app
-
-    yield
-
-    # Cleanup
-    if original_value is None:
-        os.environ.pop("MCPGATEWAY_ADMIN_API_ENABLED", None)
-    else:
-        os.environ["MCPGATEWAY_ADMIN_API_ENABLED"] = original_value
-
-    # Clear cache again for other tests
-    get_settings.cache_clear()
+@pytest.fixture
+def client(main_app_with_admin_api: FastAPI) -> TestClient:
+    return TestClient(main_app_with_admin_api, follow_redirects=False)
 
 
 def _jwt_secret() -> str:
@@ -80,14 +43,14 @@ def _jwt_secret() -> str:
     return secret.get_secret_value() if hasattr(secret, "get_secret_value") else secret
 
 
-def _build_admin_jwt(*, include_jti: bool = True, expires_in_minutes: int = 20) -> tuple[str, dict]:
+def _build_admin_jwt(*, include_jti: bool = True, expires_in_minutes: int = 20) -> tuple[str, dict[str, Any]]:
     """Build a signed admin JWT and return (token, payload).
 
     Returns:
         Tuple of (encoded JWT string, payload dict).
     """
     now = datetime.now(timezone.utc)
-    payload: dict = {
+    payload: dict[str, Any] = {
         "email": "admin@example.com",
         "exp": int((now + timedelta(minutes=expires_in_minutes)).timestamp()),
         "iat": int(now.timestamp()),
@@ -100,7 +63,7 @@ def _build_admin_jwt(*, include_jti: bool = True, expires_in_minutes: int = 20) 
 
 
 @pytest.fixture
-def disable_admin_csrf():
+def disable_admin_csrf(main_app_with_admin_api: FastAPI) -> Generator[None, None, None]:
     """Bypass admin CSRF for happy-path tests via dependency override.
 
     A separate test (``test_admin_logout_post_rejects_missing_csrf_token``)
@@ -110,17 +73,17 @@ def disable_admin_csrf():
     async def _noop():
         return None
 
-    app.dependency_overrides[enforce_admin_csrf] = _noop
+    main_app_with_admin_api.dependency_overrides[enforce_admin_csrf] = _noop
     try:
         yield
     finally:
-        app.dependency_overrides.pop(enforce_admin_csrf, None)
+        main_app_with_admin_api.dependency_overrides.pop(enforce_admin_csrf, None)
 
 
 class TestAdminLogoutTokenRevocation:
     """Happy-path: ``/admin/logout`` revokes the caller's token in the blocklist."""
 
-    def test_post_revokes_token_in_blocklist(self, disable_admin_csrf):
+    def test_post_revokes_token_in_blocklist(self, client: TestClient, disable_admin_csrf: None) -> None:
         token, payload = _build_admin_jwt()
 
         with (
@@ -132,7 +95,6 @@ class TestAdminLogoutTokenRevocation:
             mock_blocklist.revoke_token.return_value = True
             mock_get_service.return_value = mock_blocklist
 
-            client = TestClient(app, follow_redirects=False)
             response = client.post("/admin/logout", cookies={"jwt_token": token})
 
             assert response.status_code in (302, 303, 307, 200)
@@ -144,18 +106,17 @@ class TestAdminLogoutTokenRevocation:
             assert kwargs["token_expiry"] is not None
             assert kwargs["last_activity"] is not None
 
-    def test_post_without_jwt_cookie_does_not_call_blocklist(self, disable_admin_csrf):
+    def test_post_without_jwt_cookie_does_not_call_blocklist(self, client: TestClient, disable_admin_csrf: None) -> None:
         with patch("mcpgateway.services.token_blocklist_service.get_token_blocklist_service") as mock_get_service:
             mock_blocklist = MagicMock()
             mock_get_service.return_value = mock_blocklist
 
-            client = TestClient(app, follow_redirects=False)
             response = client.post("/admin/logout")
 
             assert response.status_code in (302, 303, 307, 200)
             mock_blocklist.revoke_token.assert_not_called()
 
-    def test_post_with_invalid_jwt_cookie_does_not_block_logout(self, disable_admin_csrf):
+    def test_post_with_invalid_jwt_cookie_does_not_block_logout(self, client: TestClient, disable_admin_csrf: None) -> None:
         with (
             patch("mcpgateway.admin.verify_jwt_token_cached", new_callable=AsyncMock) as mock_verify,
             patch("mcpgateway.services.token_blocklist_service.get_token_blocklist_service") as mock_get_service,
@@ -164,13 +125,12 @@ class TestAdminLogoutTokenRevocation:
             mock_blocklist = MagicMock()
             mock_get_service.return_value = mock_blocklist
 
-            client = TestClient(app, follow_redirects=False)
             response = client.post("/admin/logout", cookies={"jwt_token": "not.a.valid.jwt"})
 
             assert response.status_code in (302, 303, 307, 200)
             mock_blocklist.revoke_token.assert_not_called()
 
-    def test_post_payload_without_jti_does_not_revoke(self, disable_admin_csrf):
+    def test_post_payload_without_jti_does_not_revoke(self, client: TestClient, disable_admin_csrf: None) -> None:
         token, payload = _build_admin_jwt(include_jti=False)
         assert "jti" not in payload
 
@@ -182,13 +142,12 @@ class TestAdminLogoutTokenRevocation:
             mock_blocklist = MagicMock()
             mock_get_service.return_value = mock_blocklist
 
-            client = TestClient(app, follow_redirects=False)
             response = client.post("/admin/logout", cookies={"jwt_token": token})
 
             assert response.status_code in (302, 303, 307, 200)
             mock_blocklist.revoke_token.assert_not_called()
 
-    def test_post_blocklist_failure_does_not_block_logout(self, disable_admin_csrf):
+    def test_post_blocklist_failure_does_not_block_logout(self, client: TestClient, disable_admin_csrf: None) -> None:
         token, payload = _build_admin_jwt()
 
         with (
@@ -200,7 +159,6 @@ class TestAdminLogoutTokenRevocation:
             mock_blocklist.revoke_token.side_effect = Exception("Database unavailable")
             mock_get_service.return_value = mock_blocklist
 
-            client = TestClient(app, follow_redirects=False)
             response = client.post("/admin/logout", cookies={"jwt_token": token})
 
             assert response.status_code in (302, 303, 307, 200)
@@ -210,7 +168,7 @@ class TestAdminLogoutTokenRevocation:
 class TestAdminLogoutDenyPaths:
     """Deny-path regression tests required by AGENTS.md for security-sensitive changes."""
 
-    def test_post_with_jwt_cookie_but_no_csrf_token_is_rejected(self):
+    def test_post_with_jwt_cookie_but_no_csrf_token_is_rejected(self, client: TestClient) -> None:
         """Cookie auth + state-changing POST without a CSRF token must return 403.
 
         This protects against cross-site-forced-logout attacks. No
@@ -223,14 +181,13 @@ class TestAdminLogoutDenyPaths:
             mock_blocklist = MagicMock()
             mock_get_service.return_value = mock_blocklist
 
-            client = TestClient(app, follow_redirects=False)
             response = client.post("/admin/logout", cookies={"jwt_token": token})
 
             assert response.status_code == 403
             assert "csrf" in response.json().get("detail", "").lower()
             mock_blocklist.revoke_token.assert_not_called()
 
-    def test_get_logout_is_exempt_from_csrf(self):
+    def test_get_logout_is_exempt_from_csrf(self, client: TestClient) -> None:
         """GET /admin/logout supports OIDC front-channel logout and is exempt from CSRF.
 
         Per the OIDC Front-Channel Logout 1.0 spec the IdP issues a GET; we
@@ -247,7 +204,6 @@ class TestAdminLogoutDenyPaths:
             mock_blocklist.revoke_token.return_value = True
             mock_get_service.return_value = mock_blocklist
 
-            client = TestClient(app, follow_redirects=False)
             response = client.get(
                 "/admin/logout",
                 cookies={"jwt_token": token},

--- a/tests/unit/mcpgateway/test_admin_logout_token_revocation.py
+++ b/tests/unit/mcpgateway/test_admin_logout_token_revocation.py
@@ -17,6 +17,8 @@ regressions are caught.
 # Standard
 from datetime import datetime, timedelta, timezone
 from unittest.mock import AsyncMock, MagicMock, patch
+import importlib
+import os
 import uuid
 
 # Third-Party
@@ -28,6 +30,49 @@ from fastapi.testclient import TestClient
 from mcpgateway.admin import enforce_admin_csrf
 from mcpgateway.config import settings
 from mcpgateway.main import app
+
+
+@pytest.fixture(scope="module", autouse=True)
+def enable_admin_api():
+    """Enable admin API for logout tests.
+
+    The admin router is only registered when MCPGATEWAY_ADMIN_API_ENABLED=True.
+    This fixture sets the environment variable before the app is initialized,
+    clears the settings cache, and forces a reload of the main module to pick
+    up the setting.
+    """
+    # Store original value
+    original_value = os.environ.get("MCPGATEWAY_ADMIN_API_ENABLED")
+
+    # Set environment variable
+    os.environ["MCPGATEWAY_ADMIN_API_ENABLED"] = "true"
+
+    # Clear the settings cache so get_settings() will create a new instance
+    from mcpgateway.config import get_settings
+
+    get_settings.cache_clear()
+
+    # Force reload of config and main modules to pick up the setting
+    import mcpgateway.config
+    import mcpgateway.main
+
+    importlib.reload(mcpgateway.config)
+    importlib.reload(mcpgateway.main)
+
+    # Update the app reference in this module to use the reloaded app
+    global app
+    app = mcpgateway.main.app
+
+    yield
+
+    # Cleanup
+    if original_value is None:
+        os.environ.pop("MCPGATEWAY_ADMIN_API_ENABLED", None)
+    else:
+        os.environ["MCPGATEWAY_ADMIN_API_ENABLED"] = original_value
+
+    # Clear cache again for other tests
+    get_settings.cache_clear()
 
 
 def _jwt_secret() -> str:

